File: miraculous-backend/src/main/java/com/example/miraculousbackend/dto/AlbumTrackUploadDto.java
package com.example.miraculousbackend.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class AlbumTrackUploadDto {


    private Long artistId;
    @NotBlank(message = "Each album track must have a title")
    private String title;
    @NotNull(message = "Each album track must have a price")
    private Double price;
    private Double fullAlbumTrackVersionSize;
    private Double previewAlbumTrackVersionSize;
    private Double duration;
    @NotNull(message = "Album track must be either free or paid")
    private Boolean isFree;
}

File: miraculous-frontend/miraculous-ui/src/dto/album-track-upload-dto/album-track-upload-dto.ts
export class AlbumTrackUploadDto {

  artistId: number;
  title: string;
  price: number;
  fullAlbumTrackVersionSize: number;
  previewAlbumTrackVersionSize: number;
  duration: number;
  isFree: boolean;


  constructor(artistId?: number, title?: string, price?: number, size?: number, duration?: number, previewAlbumTrackVersionSize?: number,
              isFree?: boolean) {
    this.artistId = artistId;
    this.title = title;
    this.price = price;
    this.fullAlbumTrackVersionSize = size;
    this.duration = duration;
    this.previewAlbumTrackVersionSize = previewAlbumTrackVersionSize;
    this.isFree = isFree;
  }
}
